drivers/adc/adc_sam0.c

```diff
@@ -343,26 +343,52 @@ static int start_read(const struct device *dev,
 	switch (sequence->resolution) {
 	case 8:
 		if (sequence->oversampling) {
-			LOG_ERR("Oversampling requires 12 bit resolution");
+			LOG_ERR("Oversampling requires minimum 13 bit resolution");
 			return -EINVAL;
 		}
-
 		ADC_RESSEL(adc) = ADC_RESSEL_8BIT;
 		break;
 	case 10:
 		if (sequence->oversampling) {
-			LOG_ERR("Oversampling requires 12 bit resolution");
+			LOG_ERR("Oversampling requires minimum 13 bit resolution");
 			return -EINVAL;
 		}
-
 		ADC_RESSEL(adc) = ADC_RESSEL_10BIT;
 		break;
 	case 12:
 		if (sequence->oversampling) {
-			ADC_RESSEL(adc) = ADC_RESSEL_16BIT;
-		} else {
-			ADC_RESSEL(adc) = ADC_RESSEL_12BIT;
+			LOG_ERR("Oversampling requires minimum 13 bit resolution");
+			return -EINVAL;
+		}
+		ADC_RESSEL(adc) = ADC_RESSEL_12BIT;
```

Comment:
The ADC resolution is 12bit max.
![image](https://github.com/user-attachments/assets/ca90cba0-08e5-4991-bb61-5f7750e6a60c)

However, if you want to Accumulate, Average or Oversampling, see
![image](https://github.com/user-attachments/assets/65dacbba-08e5-4a8f-98fe-221ed5ca931b)

Check below sections:
45.6.2.9 Accumulation
45.6.2.10 Averaging
45.6.2.11 Oversampling and Decimation

This change must work for all sam0 series.

This possible it related to #76963